src/screens/ProfileScreen/elements/KarmaLock.tsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import {Text, TouchableOpacity, View} from 'react-native';
+import Tooltip from 'react-native-walkthrough-tooltip';
+import MemoIcQuestionMark from '../../../assets/icons/Ic_question_mark';
+import {normalize} from '../../../utils/fonts';
+import {LockIcon, PencilIcon} from '../../../assets';
+import {COLORS} from '../../../utils/theme';
+
+export const KarmaLock = (props: {onPressCreatePost: () => void}) => {
+  const [isTooltipShown, setIsTooltipShown] = React.useState(false);
+
+  return (
+    <View
+      style={{
+        backgroundColor: COLORS.gray100,
+        borderRadius: 12,
+        paddingVertical: 8,
+        paddingHorizontal: 4
+      }}>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The inline styles used here could be moved to a StyleSheet object at the bottom of the file. This would improve readability and performance, as the styles would only be created once instead of on every render.

```diff
- style={{
-   backgroundColor: COLORS.gray100,
-   borderRadius: 12,
-   paddingVertical: 8,
-   paddingHorizontal: 4
- }}
+ style={styles.container}
```
And add this to your StyleSheet object:
```javascript
const styles = StyleSheet.create({
  container: {
    backgroundColor: COLORS.gray100,
    borderRadius: 12,
    paddingVertical: 8,
    paddingHorizontal: 4
  },
  // other styles...
});
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->